### PR TITLE
Add Flux deployment kustomization

### PR DESCRIPTION
**Pull request by stefanprodan**
Add docs with Kustomize install instructions

### CODE REVIEW

**Comment by stefanprodan** [comment]
I think the opposite. Instead of pulling the whole Flux repo and editing files, now you can just copy/past/run the instructions, way easier and clean in my opinion.

**Comment by hiddeco** [comment]
@stefanprodan agreed, but you no longer understand what happens, just that it works.

File: docs/tutorials/get-started.md [hunk 1]
```diff
@@ -32,44 +32,83 @@ extra privileges:
...
+            - --git-poll-interval=5m
+            - --sync-interval=5m
+            - --ssh-keygen-dir=/var/fluxd/keygen
+            - --git-branch=master
+            - --git-path=namespaces,workloads
+            - --git-url=git@github.com:YOURUSER/flux-get-started
+EOF
+```
+
+We set `--git-path=namespaces,workloads` to exclude Helm manifests.
```
[the remark this answers is not in the document]
Comment (squaremo) [reply docs/tutorials/get-started.md]: On the other hand it was already in here, so I wouldn't expect it to be fixed in this PR.

File: docs/tutorials/get-started.md [hunk 2]
```diff
@@ -32,44 +32,83 @@ extra privileges:
...
-the config repository for the workloads you want Flux to deploy for
-you. You are going to need access to this repository.
+Create a kustomization file and use the Flux deploy YAMLs as base:
 
 ```sh
-$EDITOR deploy/flux-deployment.yaml
+cat > fluxcd/kustomization.yaml <<EOF
+namespace: fluxcd
+bases:
+ - github.com/fluxcd/flux//deploy
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply docs/tutorials/get-started.md]: Yes, it downloads only the deploy dir.

File: docs/tutorials/get-started.md [hunk 3]
```diff
@@ -32,44 +32,83 @@ extra privileges:
...
-Now you can go ahead and edit Flux's deployment manifest. At the very
-least you will have to change the `--git-url` parameter to point to
-the config repository for the workloads you want Flux to deploy for
-you. You are going to need access to this repository.
+Create a kustomization file and use the Flux deploy YAMLs as base:
 
 ```sh
-$EDITOR deploy/flux-deployment.yaml
+cat > fluxcd/kustomization.yaml <<EOF
+namespace: fluxcd
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply docs/tutorials/get-started.md]: I think we shouldn't encourage people to deploy things in to the default namespace. This way you can easily remove flux by deleting the `fluxcd` namespace and matches the helm install instructions.

**Comment by squaremo** [comment]
With respect to whether this is a good way to show people how to install flux for the first time:

 - I agree with Hidde that a tutorial should aim to instruct
 - I don't think using using Kustomize gets in the way of instructing, especially (or that getting people to edit files was superior on that count, necessarily). But it might need a bit of explanation of why things are being done.
 - If using Kustomize is to be preferred -- we're effectively recommending it, here -- we might need to revisit some other bits of doc as well.

File: docs/tutorials/get-started.md [hunk 4]
```diff
@@ -32,44 +32,83 @@ extra privileges:
...
+            - --manifest-generation=true
+            - --memcached-hostname=memcached.fluxcd
+            - --memcached-service=
+            - --git-poll-interval=5m
+            - --sync-interval=5m
+            - --ssh-keygen-dir=/var/fluxd/keygen
+            - --git-branch=master
+            - --git-path=namespaces,workloads
+            - --git-url=git@github.com:YOURUSER/flux-get-started
+EOF
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply docs/tutorials/get-started.md]: I've removed the sync flags and added git user and email.

File: docs/tutorials/get-started.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (squaremo) [reply docs/tutorials/get-started.md]: Yar, I am not for putting things in `default`. I must be misremembering how the port-forward works, I thought it checked the namespace `flux`.

**Comment by stefanprodan** [comment]
@squaremo @hiddeco I've moved the Kustomize install docs to a dedicated section and I've added the two example repos as next steps.

File: docs/tutorials/get-started.md [hunk 5]
```diff
@@ -146,21 +106,23 @@ paste the key there.)
...
-By default, Flux git pull frequency is set to 5 minutes.
-You can tell Flux to sync the changes immediately with:
+You can check out the Flux logs with:
 
 ```sh
-fluxctl --k8s-fwd-ns=fluxcd sync
+kubectl -n default logs deployment/flux -f
 ```
 
+The default sync frequency is 5 minutes. This can be tweaked easily.
```
Comment (hiddeco) [inline docs/tutorials/get-started.md]: Not sure if this was already there or is new, but my thought while reading this was "can be tweaked easily how?"

File: docs/tutorials/index.rst [hunk 6]
```diff
@@ -10,3 +10,4 @@ Tutorials
    get-started
    get-started-helm
    driving-flux
+   get-started-kustomize
```
Comment (hiddeco) [inline docs/tutorials/index.rst]: Would move this below `get-started-helm`

File: docs/tutorials/get-started.md (hunk 5 above)
Comment (stefanprodan) [reply docs/tutorials/get-started.md]: This was already there. Let's address it in https://github.com/fluxcd/flux/pull/2298

File: docs/tutorials/index.rst (hunk 6 above)
Comment (stefanprodan) [reply docs/tutorials/index.rst]: Done!

**Pull request merged**